core/src/main/java/io/undertow/server/handlers/accesslog/DefaultAccessLogReceiver.java

```diff
@@ -139,12 +138,14 @@ private void calculateChangeOverPoint() {
 
     @Override
     public void logMessage(final String message) {
+        if(closed) {
+            //Log handler is closing, other resources should as well, there shouldn't
+            //be resources served that required this to log stuff into AL file.
+            throw UndertowMessages.MESSAGES.failedToLogAccessOnClose();
+        }
         this.pendingMessages.add(message);
-        int state = stateUpdater.get(this);
```

Comment:
Unfortunately github picked an old coment that I thought I deleted. So, here goes my final conclusion on this: 

```
if(myIntegerField == whateverValueINeedToTest) {
    if (!myIntegerFieldUpdater.compareAndSet(whateverValueINeedToTest, newValue) {
        return;
    }
}
``` 

The one above is my preferred approach for this kind of checks. For one thing, myIntegerField being volatile is not subject to word tearing and we can safely refer to that instead of invoking myIntegerFieldUpdater.get(), so that's one less invocation. Also, when dealing with that, if the field's value is not the expected one, you exit (the exit part is not shown in the above snippet) without having to invoke the field updater, so it is cheaper.

All that being said, I am in favour of replacing it by just an invocation to compareAndSet like you did in this PR, because, in this particular run() method, for the expected use of DefaultAccessLogReceiver, the value is expected to be whateverValueINeedToTest in 99% of the times it reaches this code, so you probably were going to have to pay the price of invoking compareAndSet anyway.

Reply:
In this place its a hail marry sort of situation.  If access log is running, run(), will pick up excess messages and schedule a run, if not, this will do it. So it will either be irrelevant and needs to synced up just in case, or will schedule new run.

Reply:
Agreed. I'm marking this as resolved.